database/interface.py

```diff
@@ -345,8 +345,15 @@ def get_harvest_record_errors_by_job(self, job_id: str, **kwargs):
             .filter(HarvestRecord.harvest_job_id == job_id)
             .subquery()
         )
-        query = self.db.query(HarvestRecordError).filter(
-            HarvestRecordError.harvest_record_id.in_(select(subquery))
+        query = (
+            self.db.query(
+                HarvestRecordError,
```

Comment:
why are we returning a tuple when the interface method is `get_harvest_record_errors_by_job`? shouldn't we just be returning a list of errors?

Reply:
Instead of a list of errors with information for harvest_record_error, we need an additional field: identifier and title (source_raw) from the harvest_record table.

Reply:
If we need this data structure for this one specific call, could we call it something else, and could you put some type hinting for what it returns?

Reply:
Added a detailed description for clarification.